zvariant/src/maybe.rs

```diff
@@ -103,6 +103,49 @@ impl<'a> Maybe<'a> {
     }
 }
 
+impl Display for Maybe<'_> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        maybe_display_fmt(self, f, true)
+    }
+}
+
+pub(crate) fn maybe_display_fmt(
+    maybe: &Maybe<'_>,
+    f: &mut std::fmt::Formatter<'_>,
+    type_annotate: bool,
+) -> std::fmt::Result {
+    if type_annotate {
+        write!(f, "@{} ", maybe.full_signature())?;
+    }
+
+    let (last_inner, depth) = {
+        let mut curr = maybe.inner();
+        let mut depth = 0;
+
+        while let Some(Value::Maybe(child_maybe)) = curr {
+            curr = child_maybe.inner();
+            depth += 1;
+        }
+
+        (curr, depth)
+    };
+
+    if let Some(last_inner) = last_inner {
+        // There are no Nothings, so print out the inner
+        // value with no prefixes.
+        value_display_fmt(last_inner, f, false)?;
```

Comment:
Seems to me, this is the only place where we pass `false` for `type_annotation`. Given that it greatly complicates the code (you don't need separate crate public API if not for this), I wonder if there's a way to avoid this? Perhaps we do always print type annotation? :thinking:

Reply:
We also disable type annotation on subsequent elements after the first element on dicts and arrays. Aside from being more verbose, we would also break compatibility with `GVariant`.

Reply:
> We also disable type annotation on subsequent elements after the first element on dicts and arrays. Aside from being more verbose, we would also break compatibility with `GVariant`.

hmm.. good point.